broker/lib/controllers/ServiceFabrikApiController.js

```diff
@@ -249,45 +347,210 @@ class ServiceFabrikApiController extends FabrikBaseController {
+      ]))
+      .spread((guid, deployment) => {
+        _.set(req.body, 'trigger', trigger);
+        backupGuid = guid;
+        return this.getBackupOptions(backupGuid, deployment, req)
+          .then(backupOptions => {
+            logger.info(`Triggering backup with options: ${JSON.stringify(backupOptions)}`);
+            // Acquire read lock
+            return lockManager.lock(req.params.instance_id, {
+                lockType: CONST.ETCD.LOCK_TYPE.READ,
+                lockedResourceDetails: {
+                  resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.BACKUP,
+                  resourceType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP,
+                  resourceId: backupGuid,
+                  operation: CONST.OPERATION_TYPE.BACKUP
+                }
+              })
+              .then(() => {
+                lockedDeployment = true;
+                return eventmesh.apiServerClient.createOperation({
+                  resourceId: req.params.instance_id,
+                  operationName: CONST.OPERATION_TYPE.BACKUP,
+                  operationType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP,
+                  operationId: backupGuid,
+                  value: backupOptions
+                });
+              });
+          });
+      })
+      .then(() => {
+        backupStartedAt = new Date();
+        //check if resource exist, else create and then update
+        return Promise.try(() => eventmesh.apiServerClient.getResource('deployment', 'directors', req.params.instance_id))
+          .catch(() => eventmesh.apiServerClient.createDeployment(req.params.instance_id, {}))
+          .then(() => eventmesh.apiServerClient.updateLastOperation({
+            resourceId: req.params.instance_id,
+            operationName: CONST.OPERATION_TYPE.BACKUP,
+            operationType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP,
+            value: backupGuid
+          }));
+      })
+      .then(() => {
+        res.status(CONST.HTTP_STATUS_CODE.ACCEPTED).send({
+          name: CONST.OPERATION_TYPE.BACKUP,
+          guid: backupGuid
+        });
+      })
+      .catch(err => {
+        logger.info('Handling error :', err);
+        if (err instanceof ResourceAlreadyLocked) {
+          throw err;
+        }
+        if (lockedDeployment) {
+          return lockManager.unlock(req.params.instance_id)
+            .throw(err);
+        }
+        throw err;
+      });
+  }
+
   getLastBackup(req, res) {
+    if (config.enable_service_fabrik_v2) {
+      return this.getLastBackupV2(req, res);
+    }
+    return this.getLastBackupV1(req, res);
+  }
+
+  getLastBackupV2(req, res) {
+    return eventmesh.apiServerClient.getLastOperation({
+        resourceId: req.params.instance_id,
+        operationName: CONST.OPERATION_TYPE.BACKUP,
+        operationType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP
+      })
+      .then(backup_guid =>
```

Comment:
method getLastOperation, returning back backup_guid as response does not sound right. By the name of the method, the return should have been an operation object which should probably contain backup guid. Always keep methods/responses generic so that they can be reused.